lib/pavilion/test_config/resolver.py

```diff
@@ -688,6 +723,19 @@ def resolve_inheritance(base_config, suite_cfg, suite_path):
             suite_tests[test_cfg_name] = test_config_loader.merge(parent,
                                                                   test_cfg)
 
+            if parent is not '__base__':
+                for section in ['build', 'run']:
+                    config = test_cfg.get(section)
+                    if not config:
+                        continue
+                    new_cmd_list = []
+                    if config.get('cmds_extend_before'):
+                        new_cmd_list += config.get('cmds_extend_before')
+                    new_cmd_list += suite_tests[test_cfg_name][section]['cmds']
+                    if config.get('cmds_extend_after'):
+                        new_cmd_list += config.get('cmds_extend_after')
+                    suite_tests[test_cfg_name][section]['cmds'] = new_cmd_list
+
```

Comment:
This is what you should have in a function.

Reply:
It applies not just to inheritance, but in all steps of layering. mode -> test -> inheritance -> host -> `__base__`